src/binder/bind_node_visitor.cpp

```diff
@@ -645,6 +645,14 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::ComparisonExpression>
   sherpa_->CheckDesiredType(expr.CastManagedPointerTo<parser::AbstractExpression>());
   sherpa_->SetDesiredTypePair(expr->GetChild(0), expr->GetChild(1));
   SqlNodeVisitor::Visit(expr);
+
+  // If any of the operands are typecasts, the typecast children should have been casted by now. Pull the children up.
```

Comment:
I find it hacky to manually look for typecasts here, but we do a similar thing in select statements.

Might need to look for typecasts elsewhere, this will unblock TPC-H though I think.

In general, if your visiting needs to knock out the thing X, you need to make that change at all callers of visit(X) rather than at the implementation of visit(X) itself. Please correct me if I'm wrong.